### PR TITLE
Rename genericParameters to genericParameterClause for consistency.

### LINKED ISSUE #1678: Inconsistent naming of properties for retrieving a generic parameter clause on decl nodes
### Description

All applicable declaration nodes except `EnumDeclSyntax` use `genericParameterClause`. The respective property of `EnumDeclSyntax` is called `genericParameters`.

#### Environment 

93e905ce326457acee1d485cb0150616b92e3eae

### Steps to Reproduce

_No response_

### PR DESCRIPTION
**Pull request by TTOzzi**
Resolve #1678 

Rename `genericParameters` to `genericParameterClause` for consistency.
Since the existing `genericParameters` was declared public, we left it with the deprecated attributes. (If we don't need to keep it, I'll remove it.)

### CODE REVIEW

**Comment by TTOzzi** [comment]
While working on this, I found a few more types that use genericParameters.
I renamed them all at once, if any further modifications are needed, please let me know.

**Review by ahoppen — COMMENTED** [review]
@TTOzzi, you modified a lot of files that are generated using CodeGeneration. These changes will be reverted if you run CodeGeneration again (see https://github.com/apple/swift-syntax/blob/main/CodeGeneration/README.md). 

What you need to do instead, is to modify the declaration of these types  in the CodeGeneration package (in the CodeGeneration folder) and then run CodeGeneration as linked in the README I linked above. 

As for the deprecated compatibility functions, can you add them to  [SyntaxCompatibility.swift](https://github.com/apple/swift-syntax/blob/main/Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift)?

**Comment by TTOzzi** [comment]
> @TTOzzi, you modified a lot of files that are generated using CodeGeneration. These changes will be reverted if you run CodeGeneration again (see https://github.com/apple/swift-syntax/blob/main/CodeGeneration/README.md).
> 
> What you need to do instead, is to modify the declaration of these types in the CodeGeneration package (in the CodeGeneration folder) and then run CodeGeneration as linked in the README I linked above.
> 
> As for the deprecated compatibility functions, can you add them to [SyntaxCompatibility.swift](https://github.com/apple/swift-syntax/blob/main/Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift)?

Ah, I see! I didn't know there was such a convenient script! 
I will make the changes again. Thank you for letting me know 🙇‍♂️

File: Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift [hunk 1]
```diff
@@ -28,3 +28,175 @@ public typealias AccessPathComponentSyntax = ImportPathComponentSyntax
...
+      inheritanceClause: inheritanceClause,
+      unexpectedBetweenInheritanceClauseAndGenericWhereClause,
+      genericWhereClause: genericWhereClause,
+      unexpectedBetweenGenericWhereClauseAndMemberBlock,
+      memberBlock: memberBlock,
+      unexpectedAfterMemberBlock,
+      arena: arena
+    )
+  }
+}
```
Comment (ahoppen) [inline Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift:114]: The `Raw` don’t need to be stable. I’m adding some documentation about that in https://github.com/apple/swift-syntax/pull/1696

**Review by ahoppen — COMMENTED** [review]
Thanks. This looks good to me. One minor comment: You don’t need to provide a compatibility layer for the `RawSyntax` nodes.

Also, in case you haven’t done so already: Can you please make sure that swift-syntax builds without warnings locally?

File: Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift (hunk 1 above)
Comment (TTOzzi) [reply Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift]: I have read the SPI.md document and removed the code intended for preserving the Raw type initializer 🙂

**Comment by TTOzzi** [comment]
> Also, in case you haven’t done so already: Can you please make sure that swift-syntax builds without warnings locally?

The build is being successful without any warnings on my local. Is there anything else I should check?

**Review by ahoppen — APPROVED** [review]
Thank you

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**